### PR TITLE
Assorted toolbox fixes

### PR DESCRIPTION
**Pull request by xeno-by**
review @retronym

### CODE REVIEW

**Comment by soc** [comment]
LGTM!

One thing I have been wondering for some time ... why is it called “Toolbox” and not “Compiler”?

**Comment by xeno-by** [comment]
I'll leave this question to @odersky. Toolboxes were already there when I started my doctorate at EPFL.

File: src/reflect/scala/reflect/runtime/JavaMirrors.scala [hunk 1]
```diff
@@ -673,8 +673,10 @@ private[reflect] trait JavaMirrors extends internal.SymbolTable with api.JavaUni
         val parents = try {
           parentsLevel += 1
           val jsuperclazz = jclazz.getGenericSuperclass
-          val superclazz = if (jsuperclazz == null) AnyClass.tpe else typeToScala(jsuperclazz)
-          superclazz :: (jclazz.getGenericInterfaces.toList map typeToScala)
+          val ifaces = jclazz.getGenericInterfaces.toList map typeToScala
+          val isAnnotation = (jclazz.getModifiers & JAVA_ACC_ANNOTATION) != 0
+          if (isAnnotation) AnnotationClass.tpe :: ClassfileAnnotationClass.tpe :: ifaces
```
Comment (retronym) [inline src/reflect/scala/reflect/runtime/JavaMirrors.scala:678]: Why not just `ClassfileAnnotationClass`, given that `ClassFileAnnotation <: StaticAnnotation <:  Annotation`?

Comment (xeno-by) [inline src/reflect/scala/reflect/runtime/JavaMirrors.scala:678]: That got me wondering too, but it was like that in `ClassfileReader`, so I started thinking. And then I remembered (from my `typedParentType` refactoring) that we expect the first item in the parent list to be a class, not a trait.

**Comment by gkossakowski** [comment]
@retronym: I'd like to get your review in addition to @soc's.

**Comment by gkossakowski** [comment]
Also, should this target 2.10.x?

**Comment by xeno-by** [comment]
Why not 2.10.x? Doesn't look like it's going to break something, and it touches only an experimental API. Also, the reason for me coming up with a fix is a request from someone who needs it in 2.10.x.

**Comment by retronym** [comment]
LGTM

**Comment by xeno-by** [comment]
Ping. Can we merge this?

**Pull request merged**